### PR TITLE
Add wasmldr

### PR DESCRIPTION
**Pull request by wgwoods**
(This is the same code as enarx/enarx-keepldr#512, rebased onto this branch.)

Short summary: this PR adds [enarx-wasmldr] as an internal crate - like [shim-sgx] and [shim-sev] - so that the `enarx` binary can run WebAssembly inside a keep. Hooray! It also adds some integration tests to ensure that each of the backends can successfully run a few simple WebAssembly workloads.

The current design is a little hacky, but it was the shortest path to getting the tests working:

1. The `wasmldr` component is built into the `enarx` binary
2. It's the default target for `enarx exec` if no external binary is provided
3. `wasmldr` expects to read the `.wasm` workload in on FD3

So right now if you want to run a `.wasm` workload you do: `enarx exec 3< hello.wasm`. And this is roughly what `tests/wasmldr_tests.rs` does! (This will change in the near future to be a little less obtuse - shell redirects are black magic to most users and Rust actively sets `FD_CLOEXEC` on all opened files, so you can't pass file descriptors to child processes without resorting to `unsafe` code..)

Anyway, all the tests should pass except for 2 wasmldr integration tests involving WASI, which work with KVM but fail with SGX. Which means that these tests are working as intended, and they're already finding bugs in our shim backends (...hooray?)

See the individual commits for more detail. Thanks!

[enarx-wasmldr]: https://github.com/enarx/enarx-wasmldr/tree/819f15e
[shim-sgx]: https://github.com/enarx/enarx/tree/main/internal/shim-sgx
[shim-sev]: https://github.com/enarx/enarx/tree/main/internal/shim-sev

### CODE REVIEW

**Comment by wgwoods** [comment]
With the addition of bace07f the tests all pass now. Hooray!

File: src/main.rs [hunk 1]
```diff
@@ -151,12 +168,16 @@ fn backend(backends: &[Box<dyn Backend>]) -> &dyn Backend {
...
-    let backend = backend(backends);
+#[inline]
+fn workldr(workldrs: &[Box<dyn Workldr>]) -> &dyn Workldr {
+    // NOTE: this is stupid, but we only have one workldr, so... ¯\_(ツ)_/¯
+    &*workldrs[0]
+}
 
-    let map = mmarinus::Kind::Private.load::<mmarinus::perms::Read, _>(&opts.code)?;
+fn exec(backend: &dyn Backend, shim: impl AsRef<[u8]>, exec: impl AsRef<[u8]>) -> Result<()> {
+    //let map = mmarinus::Kind::Private.load::<mmarinus::perms::Read, _>(&opts.code)?;
```
Comment (haraldh) [inline src/main.rs:178]: does it need to stay as a comment here?

File: internal/wasmldr/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+[![Workflow Status](https://github.com/enarx/enarx-wasmldr/workflows/test/badge.svg)](https://github.com/enarx/enarx-wasmldr/actions?query=workflow%3A%22test%22)
+[![Average time to resolve an issue](https://isitmaintained.com/badge/resolution/enarx/enarx-wasmldr.svg)](https://isitmaintained.com/project/enarx/enarx-wasmldr "Average time to resolve an issue")
+[![Percentage of issues still open](https://isitmaintained.com/badge/open/enarx/enarx-wasmldr.svg)](https://isitmaintained.com/project/enarx/enarx-wasmldr "Percentage of issues still open")
```
Comment (haraldh) [inline internal/wasmldr/README.md:3]: The `cargo readme` lint in `.github/workflows/lint.yml` should be activated for wasmldr also. Along with all the other lints.

File: internal/wasmldr/src/cli.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    pub args: Vec<String>,
+}
+
+fn parse_env_var(s: &str) -> Result<(String, String)> {
+    let parts: Vec<&str> = s.splitn(2, '=').collect();
+    if parts.len() != 2 {
+        bail!("must be of the form `NAME=VAL`");
+    }
+    Ok((parts[0].to_owned(), parts[1].to_owned()))
+}
```
Comment (haraldh) [inline internal/wasmldr/src/cli.rs:42]: Can't find any usage of this function.

File: internal/wasmldr/src/workload.rs [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    // Support module-linking (https://github.com/webassembly/module-linking)
+    config.wasm_module_linking(true);
+    // module-linking requires multi-memory
+    config.wasm_multi_memory(true);
+
+    // Prefer dynamic memory allocation style over static memory
+    config.static_memory_maximum_size(0);
+    config.static_memory_guard_size(0);
+    config.dynamic_memory_guard_size(0);
+    config.dynamic_memory_reserved_for_growth(1 * 1024 * 1024);
```
Comment (haraldh) [inline internal/wasmldr/src/workload.rs:67]: We could use [`nbytes::bytes!`](https://github.com/enarx/nbytes) here.
```suggestion
    config.dynamic_memory_reserved_for_growth(bytes![1; MiB]);
```

File: src/main.rs (hunk 1 above)
Comment (wgwoods) [reply src/main.rs:178]: Heh oh yeah - I actually had a fixup patch that dropped this line.. and then I decided to leave it in just to see if any reviewer(s) would notice. So yeah, I'll make sure to delete that, and I award you with a gold star for Excellence in Code Review: ⭐

File: internal/wasmldr/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        Err(workload::Error::StringTableError) => 65,
+        Err(workload::Error::InstantiationFailed) => 65,
+        Err(workload::Error::ExportNotFound) => 65,
+        Err(workload::Error::CallFailed) => 65,
+
+        // Internal WASI errors -> EX_SOFTWARE
+        Err(workload::Error::WASIError(_)) => 70,
+
+        // General IO errors -> EX_IOERR
+        Err(workload::Error::IoError(_)) => 74,
```
[the remark this answers is not in the document]
Comment (wgwoods) [reply internal/wasmldr/src/main.rs]: done - see ca092c1.

File: internal/wasmldr/README.md (hunk 2 above)
Comment (wgwoods) [reply internal/wasmldr/README.md]: Fixed in 03bac6c.

**Pull request merged**